crates/libcontainer/src/container/builder.rs

```diff
@@ -101,11 +103,14 @@ impl<'a> ContainerBuilder<'a> {
     /// # use libcontainer::syscall::syscall::create_syscall;
     ///
     /// ContainerBuilder::new("74f1a4cb3801".to_owned(), create_syscall().as_ref())
-    /// .with_root_path("/run/containers/youki");
+    /// .with_root_path("/run/containers/youki").expect("invalid root path");
     /// ```
-    pub fn with_root_path<P: Into<PathBuf>>(mut self, path: P) -> Self {
-        self.root_path = path.into();
-        self
+    pub fn with_root_path<P: Into<PathBuf>>(mut self, path: P) -> Result<Self> {
+        self.root_path = path
+            .into()
+            .canonicalize()
+            .context("failed to canonicalize root path")?;
```

Comment:
```suggestion
            .with_context(||format!("failed to canonicalize root path {:?}", path))?;
```

Reply:
Needed to adjust it slightly as `Into<PathBuf>` is not `Debug`.